### DISCUSSION TITLE
Post body not forwarded

### DISCUSSION THREAD

**Discussion by Hanmeim** [chimurai/http-proxy-middleware#901] [Q&A]
Sry to open a new issue on this topic again. I've gone through the old issues but I really don't understand how to do it.
I'm using React 18, axios, to post requests, like this 
   `axios.post("api/blabla", {data:"blabla})`
and the post body is not forwarded to server.
I'm using http-proxy-middleware `v3.0 beta1` but the documents are strange so most of the configure file is still v2-styled.
this is my `setupProxy.js`:
```
const { legacyCreateProxyMiddleware: createProxyMiddleware, fixRequestBody } = require("http-proxy-middleware")

module.exports = function(app){ 
    app.use( 
        createProxyMiddleware('/api1',{
            logger: console,
            target: 'http://127.0.0.1:8000',
            changeOrigin:true,
            pathRewrite:{'^/api1':''},
            on:{
                ProxyReq: fixRequestBody
            }
        })
    )
}
```

**Comment by chimurai** [comment]
Could you point out the "strange" parts of the documentation?

Here you can find v3 docs and migration steps:
https://github.com/chimurai/http-proxy-middleware/discussions/768
https://github.com/chimurai/http-proxy-middleware/blob/master/MIGRATION.md

`ProxyReq`  should be `proxyReq` (camelCase)
pathRewrite is not needed anymore if you mount the proxy slightly different:

```js
const { legacyCreateProxyMiddleware: createProxyMiddleware, fixRequestBody } = require("http-proxy-middleware")

module.exports = function(app){ 
    app.use('/api1',
        createProxyMiddleware({
            logger: console,
            target: 'http://127.0.0.1:8000',
            changeOrigin:true,
            on:{
                proxyReq: fixRequestBody
            }
        })
    )
}
```

**Reply by Hanmeim** [reply]
the documents are about how to upgrade from v2 to v3. How about using v3 directly? Is there a sample of v3 style use?

**Reply by chimurai** [reply]
All the docs in Github master are for v3. (NPM documentation is still in v2, until final version of v3 is published)

https://github.com/chimurai/http-proxy-middleware/blob/master/README.md
https://github.com/chimurai/http-proxy-middleware/tree/master/recipes
